tiledb/sm/query/writers/writer_base.h

```diff
@@ -179,6 +180,14 @@ class WriterBase : public StrategyBase, public IQueryStrategy {
   /** The name of the new fragment to be created. */
   URI fragment_uri_;
 
+  /**
+   * The base name to use for a fragment.
+   *
+   * This will only be used if fragment_uri_ is empty. If no value is set, the
```

Comment:
Question, when is the fragment URI empty and when is this going to be used instead? From your test, I think I understand that we write dimension labels to existing fragments in a separate query? Why can't we use the fragment_uri parameter?

Reply:
The `fragment_uri` includes the full path to the fragment, whereas the `fragment_base_name_` is just the final part of the fragment URI (the timestamps & uuid). I want to set just the last part of the path for two distinct arrays. As mentioned in a response above, in order to use the fragment URI directly, I would have needed to leak the construction of the fragment URI to other parts of the code base.

To answer your first question, I don't think we are ever actually setting `fragment_uri` until after construction, but I wasn't sure enough of it to remove the `fragment_uri` input parameter.